### PR TITLE
Update a bunch (148+) of homepages (permanent redirect problems as found by repology)

### PR DESCRIPTION
**Pull request by samueldr**
###### Motivation for this change

This is my effort for #30636, well, a first blast at trying to fix the easy issues.

The repository was passed through a script, specific version used available here:

 * https://github.com/samueldr/repology-autofixer/commit/1730c627bc0a28990072aadc9836cf882ad1ad3c

It has changed *mostly* meta homepage URLs, **BUT**, some download URLs have been changed. This is a side-effect of the dumb (but effective) way the change is handled. All download URLs that were a sub-path of the homepages have been automatically converted. For all URLs changed, this happened to be http→https.

###### Things done

*Nothing was tested*, no build of any of the affected software was done as this is all metadata changes.

The commits have been viewed (with my eyeballs) one by one. I have checked a couple of them manually to see whether all was right or not, especially when download URLs were changed too. If something fails, it's likely that it is the download URL, and that download URL already was previously failing.

YES, if anything fails, (except for URLs having disappeared), I'll take responsibility and *clean it through the script*.

* * *

Last, I would like to know if there's already some guidelines for commits made with automated tools. This is the way my commits look (the quote quotes the problem as stated in repology):

```
zeal: Updates homepage URL.

This commit automatically fixes the following problem:

> Homepage link "http://zealdocs.org/" is a permanent redirect to
> "https://zealdocs.org/" and should be updated

(Commit automatically made using repology-autofixer)
```

If there is anything you would like added or changed to the messages, just ask, running this script is *really cheap*, it's basically one API request, and some (short) waiting time.

### CODE REVIEW

**Comment by vcunat** [comment]
@grahamc: the borg's auto-tagging has some bad edge cases, apparently, but I don't expect it matters :-)

**Comment by vcunat** [comment]
> Last, I would like to know if there's already some guidelines for commits made with automated tools.

I don't think we really have them.  The style of these commits looks good to me.  I'm not sure whether to take this way of many commits or a single big treewide commit.

**Comment by samueldr** [comment]
It is a hard choice, with regards to the contributing guidelines, whether to make a huge commit, or one commit per URL. It can be argued both ways!

 * One commit per change, using the appropriate commit naming rules. This "pollutes" the global history, but makes each change an easy to understand, logical change, with regards to the history of that file.
 * One big commit, has a more generic commit message, but does not "pollute" the global history. This does make the change way bigger, making a file's local history weirder to understand when glancing at it. One big commit makes it inconvenient to explain each change with the *exact* problems fixed.

For this particular problem fixer, the permanent redirects, it is expected that the next time the script is used, a really low number of changes will need to be done, making many commits much more palatable.

File: pkgs/development/haskell-modules/hackage-packages.nix
Comment (peti) [inline pkgs/development/haskell-modules/hackage-packages.nix]: Please don't edit auto-generated files. See the first line of this module.

**Review by peti — DISMISSED** [review]
Don't edit `hackage-packages.nix`.

**Comment by samueldr** [comment]
The `hackage-packages` file has been [blacklisted](https://github.com/samueldr/repology-autofixer/commit/4ba911f9b6aea9249099acc04bb62b831ea63c0b). I pushed the result of a new run of the tool (explaining the two screen-fulls of commits).

Just to verify, `dotnet-packages` and `python-packages` are fine to change, right? (There is no comment on those files.)

Are there other auto-generated files that such a tool would need to steer away from?

**Comment by FRidh** [comment]
`python-packages` is fine. As there are a lot of Python changes in staging currently, it may be best to base against staging or create this commit after staging is merged into master.

**Comment by samueldr** [comment]
Since it is stupidly cheap to re-run this, I prefer, for expediency's sake, to re-run on master without python-packages, then either re-run on staging with only python-packages or wait until staging is merged to re-run.

Not that there's any reason to do this fast.

**Comment by grahamc** [comment]
@vcunat w.r.t. the edge case here, right now it runs `nix-instantiate . -A stdenv`. Should I change this to `nix-instantiate . -A stdenv.out`?

**Comment by vcunat** [comment]
@grahamc: that would still be derivation.  To print the out-path, I know e.g. `nix eval -f . stdenv.outPath` (with nix-1.12).

**Comment by orivej** [comment]
I'd strongly prefer one treewide commit, or rather one commit that changes all `meta` and another that changes all `src`.

**Review by fpletz — APPROVED** [review]
I would also prefer one commit but 👍

**Pull request merged**
